### PR TITLE
#170947649 Setting up unit testing environment

### PR DESCRIPTION
**Pull request by T2Wil**
#### What does this PR do?
 This PR ensures that the back-end has all functionalities working as intended
#### Description of Task to be completed?
I should install mocha as a testing framework with chai and chai-http as assertion libraries and generate a script to run all tests written in the `/tests` folder.

#### How should this be manually tested?
- run `git clone https://github.com/andela/knights-bn-backend.git` to clone the repo
- run `git checkout ch-set-unit-testing-170947649` to switch to this branch
- run `npm i` to install dependencies
- run `npm run test` to run tests

#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[link](https://www.pivotaltracker.com/story/show/170947649)

#### Screenshots
N/A

#170947649

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
   "main": "index.js",
   "scripts": {
     "start": "nodemon --exec babel-node src/index.js",
-    "test": "test"
+    "test": "mocha --require @babel/polyfill --require @babel/register ./src/tests/*.js"
```
Comment (Niyonsengaeric) [inline package.json]: it could be better if you change the environment before running test 
ex: NODE_ENV=test

**Review by Niyonsengaeric — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (T2Wil) [reply package.json]: sounds important!
let me do it.

**Review by Kagorora — COMMENTED** [review]
Besides Eric's comment it LGTM

File: package.json (hunk 1 above)
Comment (T2Wil) [reply package.json]: @Niyonsengaeric , 
environment is now set to `test` !

**Review by Niyonsengaeric — APPROVED** [review]

**Review by Niyonsengaeric — APPROVED** [review]

**Review by Kagorora — APPROVED** [review]
LGTM

**Review by MuhanguziDavid — APPROVED** [review]
LGTM

**Pull request merged**